Refactoring to split properly utils, tray icons and GUI + optimization

Ourouk opened the pull request:
The code still needs to be tested properly, I'm using it since yesterday without issue and may need some modification . . .


_References #118 / Fixes #118_

Ourouk committed:
Avoid double use of   utils.initialiseSettings()

Ourouk committed:
Remove init theme switching from ui to main.cpp

Ourouk committed:
Transfert schedule logic to utils instead of in the UI

Ourouk committed:
fix scheduler

Ourouk committed:
separate tray icon and settings

Ourouk committed:
No settings loaded for background tray operation

MartinVonReichenberg commented on src/headers/utils.h:
```diff
@@ -19,6 +19,8 @@
 #include "src/plugins/script.h"
 #include "src/plugins/wallpaper.h"
+// Other libraries
+#include "../libraries/Scheduler.h"
```
Are relative paths generally more appropriate than the absolute ones ?

If so, why (just briefly) ?

MartinVonReichenberg commented:
Yup,
No worries.
Keep testing it either on bare metal, virtual environment and/or both . . . 😉

Ourouk committed:
Splitting tray and settings + one click switch

Ourouk commented:
@MartinVonReichenberg  I'm using it since a few weeks in the background without issues, if a lot of arch user on the git version they should be able to report bugs. I'm the most impressed of the ram usage reduction.

MartinVonReichenberg commented:
> if a lot of arch user on the git version they should be able to report bugs

_Not sure if I quite got the sentence right . . ._

Ourouk commented:
> > if a lot of arch user on the git version they should be able to report bugs
> 
> _Not sure if I quite got the sentence right . . ._

Arch Linux user can use the git version via the aur so maybe we will have return if my code caused issues... But seems stable to me

MartinVonReichenberg commented:
I might test it myself soon too . . .

Do you still consider this as `"Draft"` ?

Ourouk commented:
Not gonna lie, I'm a noob coming to open source contribution ^^, I should probably have separated the merge request in multiple smaller ones.
And did a lot of small modification on my fork:
- Separated tray icons and setting classes
- Only instance setting if needed
- Fixed multiple instantiations of the utils object, causing unnecessary ram usage
- Added one click switch theme when clicking on tray icon
- Fixed double applying theme possibility

And since I'm using this fork for a week, this doesn't seem to be a draft anymore.

MartinVonReichenberg commented:
> I should probably have separated the merge request in multiple smaller ones.

Nope, the amount of both, **Pull Requests** and **Commits** should be the amount of _ideas/wish-to-changes/wants-to-have_ changes as you already, at the time, have.

For example, this could have been a single commit when you already knew in advance that you are going to do (almost) "exactly" this amount of _wish-to-change_ ideas . . .

Ourouk commented:
Did u notice any bug ? That I would need to address pre-merge?

MartinVonReichenberg commented:
I shall look into it within the next week . . .

I also need to install SUSE, MANJARO & Kubuntu to try it out there too . . .

MartinVonReichenberg commented:
I shall look into it within the next week . . .

I also need to install SUSE, MANJARO & Kubuntu to try it out there too . . .

For more bugs, check this out:
![20250622_191042.JPG](https://github.com/user-attachments/assets/801ed376-9a71-4668-8644-6c72c0046e68)

Ourouk commented:
I did try on opensuse and tuxedo OS that was fine, there is no modification to imports so it shouldn't be impacted.

MartinVonReichenberg commented:
@Ourouk

After your commit, it looks like left mouse click button on the Koi in panel's system tray changes LIGHT/DARK theme instead of opening _(showing)_ / closing _(hiding)_ the **main Koi window** . . .

_Did not you notice that ?_

Ourouk commented:
Yeah I did write that, I should probably created a toggle for that feature. But yeah I really enjoy the quick toggle shortcut, I even tough about a shortcut for it... :-D
The motive is that for me, we're more likely to want to switch dark/light quickly than change the settings. 
I can revert it pretty quickly if u want to keep it that way on the main branch by the way will just keep my little fork :-)

MartinVonReichenberg commented:
> I should probably created a toggle for that feature. But yeah I really enjoy the quick toggle shortcut, I even tough about a shortcut for it... :-D The motive is that for me, we're more likely to want to switch dark/light quickly than change the settings.

The point is, the instant LIGHT/DARK theme switching has already been assigned to the middle mouse button (wheel click); on the contrary - it is better to leave left mouse click to be a shortcut for showing / hiding Koi's main window as this is redundant when assigning both mouse buttons for the same functionality . . .

**_See Pull Request: #86_**

Ourouk commented:
I can revert that, but wheel click is accessible when using a mouse, trackpad rip

MartinVonReichenberg commented:
> I can revert that, but wheel click is accessible when using a mouse, trackpad rip


On modern monolithic touchpads, three finger taps / clicks at once => Middle Mouse Button Click (Wheel Click)
On touchpads / trackpads with two separate physical buttons, pressing them at once => Middle Mouse Button (Wheel Click)

 Imitation of middle mouse button should be able to enable within KDE Plasma Settings . . .

Ourouk committed:
revert buttons shortcuts

Ourouk commented:
It's reverted

MartinVonReichenberg reviewed (approved):
_Seems fine by me now . . ._
